src/autotype/AutoTypePickCharsDialog.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ *  Copyright (C) 2017 Daniel Meek <https://github.com/danielmeek32>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AutoTypePickCharsDialog.h"
+
+#include <QApplication>
+#include <QDesktopWidget>
+#include <QDialogButtonBox>
+#include <QGridLayout>
+#include <QLabel>
+#include <QPushButton>
+#include <QSignalMapper>
+#include <QToolButton>
+#include <QVBoxLayout>
+
+#include "core/Config.h"
+#include "core/FilePath.h"
+
+AutoTypePickCharsDialog::AutoTypePickCharsDialog(QString string, QWidget* parent)
+    : QDialog(parent)
+    , m_string(string)
+    , m_passwordEdit(new PasswordEdit(this))
+{
+    setAttribute(Qt::WA_DeleteOnClose);
+    // Places the window on the active (virtual) desktop instead of where the main window is.
+    setAttribute(Qt::WA_X11BypassTransientForHint);
+    setWindowFlags(windowFlags() | Qt::WindowStaysOnTopHint);
+    setWindowTitle(tr("Auto-Type - KeePassXC"));
+    setWindowIcon(filePath()->applicationIcon());
+    setModal(true);
+
+    QVBoxLayout* layout = new QVBoxLayout(this);
```

Comment:
Remove all layout-specific code from this function and use a UI file instead. Only keep logic for adding buttons.